src/axom/core/ArrayBase.hpp

```diff
@@ -1158,7 +1158,7 @@ template <typename T, MemorySpace SPACE>
 struct ArrayOps
 {
 private:
-#if defined(AXOM_GPUCC) && defined(AXOM_USE_UMPIRE)
+#if defined(AXOM_USE_GPU) && defined(AXOM_GPUCC) && defined(AXOM_USE_UMPIRE)
```

Comment:
Clarification: 
Do we want/need all three guards for these functions:
```cpp
#if defined(AXOM_USE_GPU) && defined(AXOM_GPUCC) && defined(AXOM_USE_UMPIRE)
```
rather than just the following two:
```cpp
#if defined(AXOM_USE_GPU) && defined(AXOM_USE_UMPIRE)
```

Reply:
See PR #1101

Reply:
@kennyweiss  wrote, 

> Based on the discussion and Brian's comment on issue 1101, it sounds like we only need
> `#if defined(AXOM_USE_GPU) && defined(AXOM_USE_UMPIRE)`  for these guards

I agree.  @bmhan12 , what do you think?

Reply:
> I agree. @bmhan12 , what do you think?

I concur with @kennyweiss's suggestion.

Reply:
Ditto for me.

Reply:
I tried removing the guards on `AXOM_GPUCC` : this caused failures.  Sometimes ArrayBase.h gets included in a file not compiled by nvcc or hip, on a build where we have `AXOM_USE_GPU` set.  This makes the compile fail because it tries to use a CUDA type which isn't there.